pkg/restore/restore.go

```diff
@@ -1249,13 +1249,33 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
-		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
-		if err != nil {
-			ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
-			warnings.Add(namespace, err)
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	var fromCluster *unstructured.Unstructured
+
+	if !isAlreadyExistsError {
+		fromCluster, err = resourceClient.Get(name, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			objectExists = true
+		}
+
+		if err != nil && objectExists {
+			errs.Add(namespace, err)
 			return warnings, errs
 		}
+	}
+
+	if isAlreadyExistsError || objectExists {
+		// do a get call only if the prior one to check the object existence in cluster returned nil
```

Comment:
change to:
// do a get call only if we didn't run this previously

Reply:
If we ran Get before, then either we got the resource, or we errored out with the original error. In the `isAlreadyExistsError=true` case, we haven't called Get yet and need to call it now.